src/compute-client/src/controller/instance.rs

```diff
@@ -628,7 +638,8 @@ where
         let collections = arranged_logs
             .iter()
             .map(|(_, id)| {
-                let state = CollectionState::new_log_collection();
+                let state =
+                    CollectionState::new_log_collection(id.clone(), introspection_tx.clone());
```

Comment:
`CollectionIntrospection` is intended to be a more general thing in the future, involving other introspection things that will be relevant also for log collections. For example, @teskje suggested that `ComputeDependencies` could be moved under it.

Reply:
Sure, if that'll happen we can leave it as-is. But, why not make the introspection state part of `CollectionState`?

Reply:
> why not make the introspection state part of CollectionState?

It would be possible to do that, but I felt like it's better to separate these things into a struct. If it would be in `CollectionState`, then this code would be scattered throughout `controller.rs` instead of mostly being in one contiguous block.